.github/workflows/breaking-change-check.yml

```diff
@@ -0,0 +1,96 @@
+name: Breaking Change Check
+
+on:
+  # Postsubmit CI on main.
+  push:
+    branches: [main]
+  # Presubmit CI on PRs to all branches.
+  pull_request:
+    types: [labeled, unlabeled, opened, synchronize, reopened]
+  # Allows you to run this workflow manually from the Actions tab.
+  workflow_dispatch:
+
+# cancel workflow when a newer version of the workflow is triggered on the same github ref
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # get matrix for ci-jobs
+  get_matrix:
+    name: Set CI flavors
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-matrix.outputs.matrix }}
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE so job can access it
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      # Ensure node version is great enough
+      - name: Use Node.js v14.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: '14.x'
+      - id: get-matrix
+        run: echo "matrix=$(node ./common/scripts/workflow-read-matrix.mjs)" >> $GITHUB_OUTPUT
+
+  check_breaking_changes:
+    needs: get_matrix
+    name: Check Breaking Changes
+    if: ${{ github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'breaking-change') }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJSON(needs.get_matrix.outputs.matrix) }}
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE so job can access it
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      # Ensure node version is great enough
+      - name: Use Node.js v14.x
```

Comment:
14 is quite old; 18 is the LTS - can we use that instead?

Reply:
Yep, will align with what we have for CI (version 16) then we might need a fully test for all yml to bump it to 18